### DISCUSSION TITLE
`ActiveExprModel`: insert and update a model with `SimpleExpr`

### DISCUSSION THREAD

**Discussion by billy1624** [SeaQL/sea-orm#1450] [Ideas]
## Motivation

It'd be convenient if we allow inserting and updating with expression such as summation of two columns `"num1" + "num2"` or even DB built in function `CURRENT_TIMESTAMP` or `UUID()`.

## Proposed Solutions

Introduce a `SimpleExpr` version of `ActiveValue` and `ActiveModel`. Which conversion methods will be provided to convert `ActiveValue` into `ActiveExpr` and `ActiveModel` into `ActiveExprModel`.

```rs
let active_expr_model: ActiveExprModel = 
    ActiveModel {
        // ...
    }
    .into_active_expr_model();

let model: Model =
    ActiveExprModel {
        created_at: SetExpr( ... ),
    }
    .insert(db) // or `.update(db)`
    .await?;
```

**Comment by billy1624** [comment]
Comments? @tyt2y3

**Comment by tyt2y3** [comment]
I second the idea of having an ergonimc way to `SET c = c+1`.

However I think `ActiveExprModel` is not necessary.

The API can probably look like:

```rust
let change_set = ChangeSet::new(Cake).set(cake::Column::Counter, cake::Column::Counter.add(1));
change_set.insert(&db).await?; // lets pretend the expression is a function call

change_set
    .which(2) // this is the PK, basically where PK = val
    .where(SimpleExpr)
    .update(&db).await?; // if `which` or `where` is never called, we should refuse to execute
```

**Comment by tgrushka** [comment]
What about something like ... ?
```rs
ActiveValue::SetExpr(Expr::current_timestamp())
```
